torchao/_models/llama/generate.py

```diff
@@ -297,6 +296,60 @@ def main(
             dtype = _NBITS_TO_DTYPE[nbits]
             group_size = int(_quant_args[2])
             quantize_(model, uintx_weight_only(dtype, group_size, use_hqq=use_hqq))
+        elif "int8_dynamic_activation_intx_weight" in quantization:
+            from torchao.experimental.quant_api import int8_dynamic_activation_intx_weight
+            assert precision == torch.float32, "int8_dynamic_activation_intx_weight requires fp32 precision"
+
+            # Build kernels in temp location, and load them in torch
+            import glob
+            import os
+            import subprocess
+            import tempfile
+            def cmake_build_torchao_ops(temp_build_dir):
```

Comment:
is this required? why not add to torchao default build?

Reply:
is this is required, I'd suggest to put these in a util

Reply:
Happy to add it to a default build for ARM CPU if you have a code pointer?

Reply:
that would be great, here is our build code for kernels I think: https://github.com/pytorch/ao/blob/5eb6339e0b6f413c74a3dfd5e7f53449474723fc/setup.py#L53

Reply:
It doesn't look like cmake is being used there, and it would require more refactoring to use it.  I factored out the install script into a utility, though.

Reply:
OK sounds good, we can start with this for now, maybe put this in setup.py in a separate PR. but I do feel we should do that so it's easy to use, @metascroy do you want to do that in a follow up PR?

Reply:
cc @msaroufim and @atalman if any of you can provide some guidance on how we can add cmake build script to torchao/setup.py that would be great

Reply:
Ok I'll land this for now.  On creating build scripts in torchao/setup.py, I'll leave it up to you all.  We already have build scripts for these kernels in ExecuTorch and torchchat, so they are not difficult to use on those surfaces.

Reply:
but if you add build scripts to torchao, will you still need to maintain these in torchchat and executorch?

Reply:
Yes, because both executorch/torchchat need to build executorch kernels, but if we added build scripts to torchao, they'd likely only build aten kernels (unless torchao takes a dependency on executorch).

Reply:
OK, so looks like it's the same kernel implementation, but different build script for torchao/executorch/torchchat in this case

Reply:
Example in et for cmake https://github.com/pytorch/executorch/blob/main/setup.py#L496